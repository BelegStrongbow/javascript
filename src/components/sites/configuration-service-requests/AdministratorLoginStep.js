import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { defineMessages, FormattedMessage } from "react-intl";
import { LargeButton, makeButtonFullWidth, LargeSecondaryButton } from "../../Button";
import ButtonsContainer from "../../general/ButtonsContainer";

let messages = defineMessages( {
	administratorLoginRequired: {
		id: "requestConfiguration.administratorLoginRequired",
		defaultMessage: "We will, of course, need an administrator login to your site. Please create a new admin user on your site with the email address install@yoast.com.",
	},
	administratorLoginConfirmation: {
		id: "requestConfiguration.administratorLoginConfirmation",
		defaultMessage: "I've created a new admin user with the email address install@yoast.com.",
	},
} );

const StyledLabel = styled.label`
	margin-left: 12px;
`;

const WideLargeButton = makeButtonFullWidth( LargeButton );
const WideSecondaryButton = makeButtonFullWidth( LargeSecondaryButton );

class AdministratorLoginStep extends React.Component {
	constructor( props ) {
		super( props );

		this.state = {
			confirmed: props.confirmed,
		};

		this.toggleConfirmed = this.toggleConfirmed.bind( this );
		this.handleContinue = this.handleContinue.bind( this );
	}
	/**
	* Toggles the checkbox to the administrator login information was sent.
	*
	* @param {object} event The event of toggeling the administrator login information checkbox.
	*
	* @returns {void}
	*/
	toggleConfirmed( event ) {
		if ( this.state.confirmed ) {
			this.props.resetStep();
		}

		this.setState( {
			confirmed: event.target.checked,
		} );
	}
	/**
	 * Handles the continue button, when the administrator login information is confirmed the corresponding step is completed,
	 * and holds the user input in the props.
	 *
	 * @returns {void}
	 */
	handleContinue() {
		if ( ! this.state.confirmed ) {
			return;
		}

		this.props.onSubmit( this.state );
		this.props.completeStep();
	}

	render() {
		return (
			<div>
				<p>
					<FormattedMessage
						id={ messages.administratorLoginRequired.id }
						defaultMessage={ messages.administratorLoginRequired.defaultMessage } />
				</p>
				<p>
					<input
						id="administratorLoginConfirmation"
						type="checkbox"
						onChange={ this.toggleConfirmed }
						checked={ this.state.confirmed } />
					<StyledLabel htmlFor="administratorLoginConfirmation">
						<FormattedMessage
							id={ messages.administratorLoginConfirmation.id }
							defaultMessage={ messages.administratorLoginConfirmation.defaultMessage } />
					</StyledLabel>
				</p>
				<ButtonsContainer>
					<WideSecondaryButton onClick={ this.props.onClose } >
						<FormattedMessage id="requestConfiguration.close" defaultMessage="close"/>
					</WideSecondaryButton>
					<WideLargeButton
						onClick={ this.handleContinue }
						type="submit"
						enabledStyle={ this.state.confirmed }
					>
						<FormattedMessage id="requestConfiguration.continue" defaultMessage="continue"/>
					</WideLargeButton>
				</ButtonsContainer>
			</div>
		);
	}
}

AdministratorLoginStep.propTypes = {
	confirmed: PropTypes.bool.isRequired,
	onSubmit: PropTypes.func.isRequired,
	onClose: PropTypes.func.isRequired,
	completeStep: PropTypes.func,
	resetStep: PropTypes.func,
};

export default AdministratorLoginStep;
