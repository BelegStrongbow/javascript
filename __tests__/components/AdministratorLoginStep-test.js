import React from 'react';
import AdministratorLoginStep from "../../src/components/sites/configuration-service-requests/AdministratorLoginStep";
import { createComponentWithIntl } from "../../utils";

test('The AdministratorLoginStep component matches the snapshot', () => {
	const component = createComponentWithIntl( <AdministratorLoginStep
				confirmed={ true }
				onSubmit={ () => { console.log( "clicked on Continue" ) } }
				onClose={ () => { console.log( "clicked on Close" ); } }
				completeStep={ () => { console.log( "clicked on Continue" ); } }
				resetStep={ () => { console.log( "Unchecked the confirmation" ); } }/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});