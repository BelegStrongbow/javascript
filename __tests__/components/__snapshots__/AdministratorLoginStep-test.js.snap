// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AdministratorLoginStep component matches the snapshot 1`] = `
.c4 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  border-bottom: 2px solid rgba(0,0,0,0.2);
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c4:hover,
.c4:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c4:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c7 {
  text-shadow: 0px 0px 2px #000;
}

.c3 {
  color: #000;
  box-shadow: none;
  background-color: #fff;
  border: 1px solid #000;
  min-width: 152px;
}

.c3:hover,
.c3:focus {
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border: 0;
  text-shadow: 0px 0px 2px #000;
}

.c6 {
  min-width: 152px;
}

.c1 {
  -webkit-flex: 1 0 200px;
  -ms-flex: 1 0 200px;
  flex: 1 0 200px;
  padding-bottom: 16px;
  text-align: right;
}

.c1 a,
.c1 button {
  margin-left: 12px;
}

.c0 {
  margin-left: 12px;
}

@media screen and (max-width:800px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c1 a,
  .c1 button {
    margin-left: 0;
    margin-bottom: 8px;
  }
}

@media screen and ( max-width:800px ) {
  .c5 {
    width: 100%;
    background-image: none;
    padding: 12px 16px;
  }
}

@media screen and ( max-width:800px ) {
  .c2 {
    width: 100%;
    background-image: none;
    padding: 12px 16px;
  }
}

<div>
  <p>
    <span>
      We will, of course, need an administrator login to your site. Please confirm you have delivered this to us.
    </span>
  </p>
  <p>
    <input
      checked={true}
      id="administratorLoginConfirmation"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="c0"
      htmlFor="administratorLoginConfirmation"
    >
      <span>
        I've created a new admin user with the email address install@yoast.com and send the details to that email address.
      </span>
    </label>
  </p>
  <div
    className="c1"
  >
    <button
      className="c2 c3 c4"
      onClick={[Function]}
      type="button"
    >
      <span>
        close
      </span>
    </button>
    <button
      className="c5 c6 c7 c4"
      onClick={[Function]}
      type="submit"
    >
      <span>
        continue
      </span>
    </button>
  </div>
</div>
`;
